lib/private/AppConfig.php

```diff
@@ -30,413 +33,1061 @@
+	 * @see setValueString()
+	 * @see setValueFloat()
+	 * @see setValueBool()
+	 * @see setValueArray()
+	 */
+	public function setValueInt(
+		string $app,
+		string $key,
+		int $value,
+		bool $lazy = false,
+		bool $sensitive = false
+	): bool {
+		return $this->setTypedValue(
+			$app,
+			$key,
+			(string)$value,
+			$lazy,
+			self::VALUE_INT | ($sensitive ? self::VALUE_SENSITIVE : 0)
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param float $value config value
+	 * @param bool $lazy set config as lazy loaded
+	 * @param bool $sensitive if TRUE value will be hidden when listing config values.
+	 *
+	 * @return bool TRUE if value was different, therefor updated in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy loading
+	 * @see setValueString()
+	 * @see setValueInt()
+	 * @see setValueBool()
+	 * @see setValueArray()
+	 */
+	public function setValueFloat(
+		string $app,
+		string $key,
+		float $value,
+		bool $lazy = false,
+		bool $sensitive = false
+	): bool {
+		return $this->setTypedValue(
+			$app,
+			$key,
+			(string)$value,
+			$lazy,
+			self::VALUE_FLOAT | ($sensitive ? self::VALUE_SENSITIVE : 0)
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param bool $value config value
+	 * @param bool $lazy set config as lazy loaded
+	 *
+	 * @return bool TRUE if value was different, therefor updated in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy loading
+	 * @see setValueString()
+	 * @see setValueInt()
+	 * @see setValueFloat()
+	 * @see setValueArray()
+	 */
+	public function setValueBool(
+		string $app,
+		string $key,
+		bool $value,
+		bool $lazy = false
+	): bool {
+		return $this->setTypedValue(
+			$app,
+			$key,
+			(string)$value,
+			$lazy,
+			self::VALUE_BOOL
+		);
+	}
+
+	/**
+	 * @inheritDoc
+	 *
+	 * @param string $app id of the app
+	 * @param string $key config key
+	 * @param array $value config value
+	 * @param bool $lazy set config as lazy loaded
+	 * @param bool $sensitive if TRUE value will be hidden when listing config values.
+	 *
+	 * @return bool TRUE if value was different, therefor updated in database
+	 * @since 29.0.0
+	 * @see IAppConfig for explanation about lazy loading
+	 * @see setValueString()
+	 * @see setValueInt()
+	 * @see setValueFloat()
+	 * @see setValueBool()
+	 */
+	public function setValueArray(
+		string $app,
+		string $key,
+		array $value,
+		bool $lazy = false,
+		bool $sensitive = false
+	): bool {
+		try {
+			return $this->setTypedValue(
+				$app,
+				$key,
+				json_encode($value, JSON_THROW_ON_ERROR),
+				$lazy,
+				self::VALUE_ARRAY | ($sensitive ? self::VALUE_SENSITIVE : 0)
+			);
+		} catch (JsonException $e) {
+			$this->logger->warning(
+				'could not setValueArray', ['app' => $app, 'key' => $key, 'exception' => $e]
+			);
 		}
```

Comment:
```suggestion
		} catch (JsonException $e) {
			$this->logger->error(
				'could not setValueArray', ['app' => $app, 'key' => $key, 'exception' => $e]
			);
			throw $e;
		}
```

This is too important. It basically means the function did not operate at all. This should not go silent. App authors need to change their code immediately, so this should log as error and throw hard.